### ISSUE TITLE
Screensaver with "wake up with camera" feature

### ISSUE THREAD

**Issue by anordvall** [igorgladkov/rotini#38]
Like in Imperihome it would be cool & useful if you can configure Rotini to start a screensaver after a certain amount of inactivity. It would be good if the screensaver could be either completely black or configured to for instance be a specific sitemap (e.g. consisting of a full screen clock)

To wake the app from the screensaver you could either click on the screen or make a movement in front of the device camera.

The sensitivity of the "wake up with camera" functionality should be configurable

Possible use cases:
- wall mounted tablet that wakes the app when you walk up to it
- tablet on a bedside table that shows a clock and wakes up when you move your hand in front of it

For the bedside table use case it would be good if the brightness of the clock (or screen) can be configured.

**Comment by igorgladkov** [comment]
Did you consider use build-in Android day dreaming feature? It wouldn't show any sitemap, but you can show clock or something.
I actually did some development here before to allow more cool stuff than day dreaming gets you, but the main issue was that some Android devices don't have good proximity sensors (which is simpler solution), while doing it with camera is more involved, so it is on hold for now.

PS Ecobee 3 thermostat did that screensaver/motion detection thing nicely

**Comment by anordvall** [comment]
The tablet (Huawei MediaPad) I just bought for using on bedside tables doesn't seem to support daydream :( Also, as you say the proximity sensors are usually quite bad... you need to wave your hand just a few cm from the screen while the wake up with camera feature in Imperihome works really well.

It's ok if you have this enhancement suggestion on hold for now. Maybe something to pick up later once the final version is out and if you have some spare time :)

**Comment by d0t101101** [comment]
All points taken here - I realize this feature enhancement is somewhat back-burnered, but I hope to see it in the future.  My desired use case - display infrastructure charts in a rolling playlist during a set schedule, and blank the screen during off hours.

This would make the wall mounted tablets do something semi-useful even when not in use!

**Comment by anordvall** [comment]
Perhaps displaying e.g. a clock and a temperature on the screen (maybe sometime moving a bit)? Would be useful as time & outdoor temperature is good to know :)

A somewhat related question; Would it be possible to make Rotini trigger Tasker when a new openHAB event occurs? Example of a Tasker config: if motion is detected in the livingroom through Rotini, then wake up the device.

Even better, maybe that could also be implemented in the client? If a specific widget updates, then wake up from the Rotini screensaver!

**Comment by MrMontesa** [comment]
I'm running a wall mountedd tab with Motion detector & Tasker.
Motion detector recognizes movement in front of the camera. Tasker is able to pickup this information and turns on the display for a minute. After that it backs out the screen until next movement is recognized. Works like charm.

**Comment by anordvall** [comment]
Unfortunately none of my tablets have a proximity sensor :(

@MrMontesa, how close to the tablet do you need to be for it to wake up? I would really like it to wake up when you are still a couple of meters away from it so it is fully awake and ready while you are standing next to it.

**Comment by igorgladkov** [comment]
There is also a possibility to create a setup where external motion sensor is used and then server tells the client to wake up

**Comment by anordvall** [comment]
@igorgladkov, do you mean that is possible today, or would it need a feature request?

Also, do you mean it would wake the client from it's own screensaver, or would it wake the client from the device screensaver?

**Comment by igorgladkov** [comment]
It would need a new feature, just saying that I was thinking of possibilities.
I would prefer to have some kind of client screensaver ideally, so it could be configured with some widgets like weather/thermostat/etc

**Comment by anordvall** [comment]
I agree with regards to a client screensaver. I would also really like to be able to wake it up based on a widget (e.g. motion sensor) updating. Should we perhaps close this requrest and collect all screensaver wishes in a new issue report?

**Comment by igorgladkov** [comment]
Let's just keep this one open

**Comment by MrMontesa** [comment]
Hey @anordvall,
my device wakes up 1,5 meter when walking towards the device. I've removed lock screen which instantly opens the app. Works like charm. My corridor where the tablet is mounted suits that setup best. Also, I've tweeked Motion Detector app for days. At daytime it awakes 1,5  meters. At nighttime it activates when any light is light up.
External motion sensor also works nicely but was not needed in my setup. Could be more reliable for @anordvall when "long distance" activation is needed.
A dedicated feature within Rotini would be neat, but since there are other ways to achive this, it should be low prio.

**Comment by anordvall** [comment]
@MrMontesa What's the name of the Motion Detector App you are referring to? Are you using the camera or the proximity sensor?

**Comment by MrMontesa** [comment]
My Nexus 7 (2012) only has front cam. no proximity sensor.

**Comment by anordvall** [comment]
Ok cool.
Is the app perhaps this one? https://play.google.com/store/apps/details?id=org.motion.detector

**Comment by MrMontesa** [comment]
Yes, thats the app. There are several howto's out there. Just google for "wallmounted tablet". I used a [german howto](http://hausautomation.stern-av.de/2015/11/homematic-android-tablet-als-hausautomations-front-end-einsetzen/) but guess there are others as well.

**Comment by anordvall** [comment]
I've looked around but have decided to wait until @igorgladkov implements an internal app screen saver. Igor, I'm having big hopes that you can add an option to wake up from the screensaver based on a specific widget update, which would make it possible for me to wake it up when one of my Z-wave motion detectors on ground floor detects motion. This will be by far the cleanest solution IMHO.

**Comment by frode** [comment]
I have a tablet framed in a wooden picture frame, that function as a picture frame and smart home remote / information display.

When the device has been idle for x-minutes, it should go into a screensaver mode that displays a picture slideshow from either device memory/sdcard or other providers that could be extended later on (LAN, dropbox, google+, flickr and so on).

Could also be nice to display some optional info on top of the pictures. 
Example: 
clock/date, weather forecast.

**Comment by frode** [comment]
@anordvall Just a heads up on your idea to wake up the tablet when motion has been detected by example a z-wave device, if you have the tablet connected to a fibaro wallplug or similar, and set up the tablet to wake up when connected to mains, you should already be able to achieve this :+1:

**Comment by anordvall** [comment]
@frode Thanks for your suggestion! Yes I've already played around with my tablet and a Fibaro wall plug some days ago but I could not get it to work as I was hoping for. It seems the screen turns on also when the Fibaro wall plug shuts off. Also if motion occurs during the time the screen is on, I don't want it to go to sleep but rather reset the screen saver delay back to 0. Did you already solve this somehow? Btw, I want to do this without rooting the device :)

**Comment by frode** [comment]
@anordvall Just got Rotini installed on one of my tablets yesterday, put it inside a Ikea picture frame, hopefully I'll get a permanent spot for it soon, then I will play on a wakeup solution for it. I believe it should be possible to wakeup a Android tablet without root permission.

**Comment by anordvall** [comment]
@frode Nice, you should upload some screenshots on Google+ to inspire the rest of us :)

To clarify, I can get the tablet to wake up without root, by using some "hack" found here:
http://android.stackexchange.com/posts/108443/revisions

The problem is that the tablet turns on also when the wallplug turns off, and not only when the plug turns on. I tried using the "Event -> Power -> Battery Changed" for this. I guess I can set the wallplug to turn off before the tablet gets a chance to fall asleep, but it doesn't feel like a solid solution :)

**Comment by igorgladkov** [comment]
Clock screensaver will be support in 0.8 (timeout or item state based). Screensaver functionality will be expanded more in the future.
